dspace-spring-rest/src/test/java/org/dspace/app/rest/BrowsesResourceControllerIT.java

```diff
@@ -498,4 +499,226 @@ public void testPaginationBrowseByDateIssuedItems() throws Exception {
+                                       )))
+                   //Verify startsWith parameter is included in the links
+                    .andExpect(jsonPath("$._links.self.href", containsString("?startsWith=U")));
+
+        //** WHEN **
+        //An anonymous user browses the entries in the Browse by Author endpoint
+        //with startsWith set to T and scope set to Col 1
+        getClient().perform(get("/api/discover/browses/author/entries?startsWith=T")
+                                .param("scope", col1.getID().toString()))
+
+                   //** THEN **
+                   //The status has to be 200 OK
+                   .andExpect(status().isOk())
+                   //We expect the content type to be "application/hal+json;charset=UTF-8"
+                   .andExpect(content().contentType(contentType))
+
+                   //We expect only the entry "Turing, Alan Mathison" to be present
+                   .andExpect(jsonPath("$.page.totalElements", is(1)))
+                   //As entry browsing works as a filter, we expect to be on page 0
+                   .andExpect(jsonPath("$.page.number", is(0)))
+
+                   //Verify that the index filters to the "Turing, Alan'" items.
+                   .andExpect(jsonPath("$._embedded.browseEntries",
+                                       contains(BrowseEntryResourceMatcher.matchBrowseEntry("Turing, Alan Mathison", 1)
+                                       )))
+                   //Verify that the startsWith paramater is included in the links
+                    .andExpect(jsonPath("$._links.self.href", containsString("?startsWith=T")));
+
+        //** WHEN **
+        //An anonymous user browses the entries in the Browse by Subject endpoint
+        //with startsWith set to C
+        getClient().perform(get("/api/discover/browses/subject/entries?startsWith=C"))
+
+                   //** THEN **
+                   //The status has to be 200 OK
+                   .andExpect(status().isOk())
+                   //We expect the content type to be "application/hal+json;charset=UTF-8"
+                   .andExpect(content().contentType(contentType))
+
+                   //We expect only the entry "Computing" to be present
+                   .andExpect(jsonPath("$.page.totalElements", is(1)))
+                   //As entry browsing works as a filter, we expect to be on page 0
+                   .andExpect(jsonPath("$.page.number", is(0)))
+
+                   //Verify that the index filters to the "Computing'" items.
+                   .andExpect(jsonPath("$._embedded.browseEntries",
+                                       contains(BrowseEntryResourceMatcher.matchBrowseEntry("Computing", 3)
+                                       )))
+                   //Verify that the startsWith paramater is included in the links
+                    .andExpect(jsonPath("$._links.self.href", containsString("?startsWith=C")));
+
+        // ---- BROWSES BY ITEM ----
```

Comment:
it is better to create separate test for that, so that a failure will be more easily to understand

Reply:
Separated tests for item and entry browses.